src/ripple/rpc/impl/Handler.cpp

```diff
@@ -205,36 +245,47 @@ class HandlerTable
             version > (betaEnabled ? RPC::apiBetaVersion
                                    : RPC::apiMaximumSupportedVersion))
             return nullptr;
-        auto i = table_.find(name);
-        return i == table_.end() ? nullptr : &i->second;
+
+        auto const range = table_.equal_range(name);
+        auto const i = std::find_if(
+            range.first, range.second, [version](auto const& entry) {
+                return version >= entry.second.minApiVer_ &&
+                    version <= entry.second.maxApiVer_;
+            });
+
+        return i == range.second ? nullptr : &i->second;
     }
 
     std::vector<char const*>
     getHandlerNames() const
     {
         std::vector<char const*> ret;
-        ret.reserve(table_.size());
         for (auto const& i : table_)
-            ret.push_back(i.second.name_);
+        {
+            // Note, table_ is always ordered, allowing such a simple check
+            if (ret.empty() || std::strcmp(ret.back(), i.second.name_) != 0)
```

Comment:
Can we return a `std::set<char const*>` instead of manually checking for duplicates in the vector?

Reply:
I've made the necessary changes for using a `std::set` in at the tip of this [branch](https://github.com/ckeshava/rippled/tree/handlerSet)

feel free use it. this is the [commit hash](b705fc4a3c0ce73e6e93480134e1027ead34ab17)

Reply:
thank you, will grab your commit - agree that's an improvement.